pyccl/pk2d.py

```diff
@@ -217,7 +271,7 @@ def apply_halofit(self, cosmo, pk_linear=None):
                                "redshifts. If using the calculator mode, "
                                "check the support of the background data.")
 
-        pk2d = Pk2D(empty=True)
+        pk2d = CCLObject.__new__(Pk2D)
```

Comment:
Can we make this `Pk2D.__new__(Pk2D` just in case we add a custom `__new__` method in the future?

Reply:
We'll realise then if that happens.

Reply:
Don't bet on it, usage of `__new__` is kind of obscure and not everyone knows what it does. I am just pointing out a potential future bug, if you think it's not worth it, resolve.

Reply:
Done